src/MICore/LaunchOptions.cs

```diff
@@ -211,8 +211,36 @@ public bool ShouldStartServer()
 
         static internal LocalLaunchOptions CreateFromXml(Xml.LaunchOptions.LocalLaunchOptions source)
         {
+            string miDebuggerPath = source.MIDebuggerPath;
+
+            // If no path to the debugger was specified, look for the proper binary in the user's $PATH
+            if (String.IsNullOrEmpty(miDebuggerPath))
+            {
+                string debuggerBinary = null;
+                switch (source.MIMode)
+                {
+                    case Xml.LaunchOptions.MIMode.gdb:
+                        debuggerBinary = "gdb";
+                        break;
+
+                    case Xml.LaunchOptions.MIMode.lldb:
+                        debuggerBinary = "lldb-mi";
+                        break;
+                }
+
+                if (!String.IsNullOrEmpty(debuggerBinary))
+                {
+                    if (PlatformUtilities.IsWindows())
+                    {
+                        Path.ChangeExtension(debuggerBinary, "exe");
+                    }
```

Comment:
Removed the Windows check entirely - all scenarios on Windows currently specify an explicit path anyway.